### DISCUSSION TITLE
Has anyone used the RenderIfDefined module?

### DISCUSSION THREAD

**Discussion by matthewmcgarvey** [luckyframework/lucky#1479] [Q&A] [answered]
Lucky's page include this method called `render_if_defined` that I bet you've never heard of. https://github.com/luckyframework/lucky/blob/79e7108a807ccd8573b5ead3b7bd418051c05c72/src/lucky/page_helpers/render_if_defined.cr#L1-L7

Has anyone actually used it? It seems like a problem that has better solutions. I'm thinking about removing it

**Comment by stephendolan** [comment]
I actually didn't know it was available until the latest "Learn" PR for the Lucky Website where Jeremy used it!

I do think there are some great applications for things like help messages or tutorial-style content, and I want to experiment with a generic modal container that I could only render if the page has `modal_content` defined. Right now I throw the full modal markup at the bottom of a page that needs it and hide it, but this seems like it could be way more elegant.

**Comment by jwoertink** [comment, marked as answer]
Yeah, we use it in our apps. It's also used in the [Lucky website](https://github.com/luckyframework/website/blob/6785b7a8682d1cb12b107586a618361527e662d6/src/pages/page_layout.cr#L23). We have it documented as well https://luckyframework.org/guides/frontend/rendering-html#using-render_if_defined-in-layouts-for-optional-content

I haven't seen any issues with this, is there something specific you're seeing that we should change?

**Reply by matthewmcgarvey** [reply]
Nah that's fine. I was going through the html page code, saw that module and could find anything using it so I figured it was extra code that could be taken out
